### PR TITLE
`Approximate<M>` privacy measure

### LINKED ISSUE #1737: represent delta terms in privacy measures via `Approximate<M: Measure>`
As we continue to support more measures of privacy, in particular approximate-zCDP, it will be useful to separate the delta term in the privacy measure.

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes #1737

Instead of getting into a situation where we need one new privacy measure for each of MaxDivergence, ZeroConcentratedDivergence, RenyiDivergence, and any other measure of privacy (maybe bounded range?), we can simply introduce an Approximate<M> privacy measure. This will also allow us to write combinators that are generic over all δ-approximate definitions.

### CODE REVIEW

**Comment by Shoeboxam** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opendp/opendp/1768?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1768** <a href="https://app.graphite.dev/github/pr/opendp/opendp/1768?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1716** <a href="https://app.graphite.dev/github/pr/opendp/opendp/1716?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#1767](https://github.com/opendp/opendp/pull/1767) <a href="https://app.graphite.dev/github/pr/opendp/opendp/1767?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1760** <a href="https://app.graphite.dev/github/pr/opendp/opendp/1760?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1703** <a href="https://app.graphite.dev/github/pr/opendp/opendp/1703?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#1715](https://github.com/opendp/opendp/pull/1715) <a href="https://app.graphite.dev/github/pr/opendp/opendp/1715?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#1732](https://github.com/opendp/opendp/pull/1732) <a href="https://app.graphite.dev/github/pr/opendp/opendp/1732?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Shoeboxam and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: rust/src/measures/ffi.rs [hunk 1]
```diff
@@ -116,16 +122,73 @@ pub extern "C" fn opendp_measures__smoothed_max_divergence(
...
+/// ### `d`-closeness
+/// For any two data sets $x, x' \in \texttt{D}$ and any $d$ of type $(\texttt{MO::Distance}, \texttt{MO::Distance})$,
+/// where $d = (d', \delta)$,
+/// we say that $M(x), M(x)$ are $d$-close under the $\delta$-approximate $MO$-DP (abbreviated as $D_{MO}^{\delta}$)
+/// when there exist events $E$ (depending on $M(x)$) and $E'$ (depending on $M(x')$)
+/// such that $\Pr[E] \ge 1 - \delta$, $\Pr[E'] \ge 1 - \delta$, and
+///
+/// ```math
+/// D_{MO}^{\delta}(M(x)|_E \|\| M(x')|_{E'}) = D_{MO}(M(x)|_E \|\| M(x')|_{E'})
+/// ```
```
Comment (mccalluc) [inline rust/src/measures/ffi.rs:143]: This is probably causing the CI failure:
```
/home/runner/work/opendp/opendp/python/src/opendp/measures.py:
docstring of opendp.measures.approximate:21:
WARNING: Pygments lexer name 'math' is not known
```

**Review by mccalluc — CHANGES_REQUESTED** [review]
I'll give this a closer look, but one immediate problem with the docs build.

**Review by mccalluc — COMMENTED** [review]
Otherwise, looks ok. Let me know if there's anything to look at closely.

**Review by mccalluc — APPROVED** [review]
Looks good.

File: R/opendp/vignettes_old/comb.Rmd [hunk 2]
```diff
@@ -235,16 +235,16 @@ composed_fixed = make_basic_composition(
...
 
-#def test_make_pureDP_to_fixed_approxDP():
+#def test_make_approximate():
 #    meas = make_basic_composition([
-#        make_pureDP_to_fixed_approxDP(make_laplace(10.)),
+#        make_approximate(make_laplace(10.)),
 #        make_fix_delta(make_zCDP_to_approxDP(make_base_gaussian(10.)), delta=1e-6)
 #    ])
 #
 #    print(meas.map(1.))
```
[the remark this answers is not in the document]
Comment (Shoeboxam) [reply R/opendp/vignettes_old/comb.Rmd:244]: Yep. Might as well update it if relevant. Best to just remove though.

**Pull request merged**